main/parse.c

```diff
@@ -3446,6 +3446,9 @@ extern void createTags(unsigned char *buffer, size_t bufferSize,
 	finiParserTrashBox ();
 	teardownAnon ();
 	teardownWriter(fileName);
+
+	if (buffer)
+		mio_free (mio);
```

Comment:
Is this a good idea? Inside createTagsWithFallback() the function openInputFile() is called in which:

1. Previously assigned File.mio is freed using mio_free()
2. The passed mio is assigned to File.mio

This means that when you free mio like this, the next time this function is called and openInputFile() is called, File.mio will contain an invalid pointer and will perform mio_free() on the invalid pointer.

Maybe it would be a good idea to add a loop in the demo app so the buffer is parsed several times because I think with this change it would reliably crash then.

Reply:
When we introduced mio from geany to ctags. I didn't consider that we will make libctags and link it to Geany. So I made various changes to mid of ctags.

Reference counting is one of such changes.

openInputFile:
```
	File.mio = mio? mio_ref (mio): getMio (fileName, openMode, memStreamRequired);
```

mio_ref increments the counter of mio.
mio_free decrements the counter of mio.

So we can avoid the crash.

Do you think we should change the name to mio_unref?

Reply:
Oh I see. I assumed free() frees MIO regardless of the reference count. Calling it _unref() might really be better.